File: src/main/java/de/pflugradts/passbird/application/commandhandling/command/AbstractSingleCharInputCommand.java
package de.pflugradts.passbird.application.commandhandling.command;

import de.pflugradts.passbird.domain.model.transfer.Input;

public abstract class AbstractSingleCharInputCommand extends AbstractInputCommand {

    private static final int MAX_COMMAND_SIZE = 1;

    protected AbstractSingleCharInputCommand(final Input input) {
        super(input);
        if (input.getCommand().getSize() > MAX_COMMAND_SIZE) {
            throw new IllegalArgumentException("parameter for command '" + input.getCommand().getChar(0)
                + "' not supported: " + input.getCommand().slice(1).asString());
        }
    }

}

File: src/main/kotlin/de/pflugradts/passbird/application/commandhandling/command/AbstractSingleCharInputCommand.kt
package de.pflugradts.passbird.application.commandhandling.command

import de.pflugradts.passbird.domain.model.transfer.Input

private const val MAX_COMMAND_SIZE = 1

abstract class AbstractSingleCharInputCommand protected constructor(val input: Input) : AbstractInputCommand(input) {
    init {
        require(input.command.size <= MAX_COMMAND_SIZE) {
            ("parameter for command '${input.command.getChar(0)}' not supported: ${input.command.slice(1).asString()}")
        }
    }
}
